plugins/modules/purefa_vg.py

```diff
@@ -572,40 +682,71 @@ def update_vgroup(module, array):
 
 def recover_vgroup(module, array):
     """Recover Volume Group"""
+    api_version = array.get_rest_version()
     changed = True
     if not module.check_mode:
-        try:
-            array.recover_vgroup(module.params["name"])
-        except Exception:
+        if LooseVersion(PRIORITY_API_VERSION) <= LooseVersion(api_version):
+            res = array.patch_volume_groups(
+                names=[module.params["name"]],
+                context_names=[module.params["context"]],
+                volume_group=VolumeGroupPatch(destroyed=False),
+            )
+        else:
+            res = array.patch_volume_groups(
+                names=[module.params["name"]],
+                volume_group=VolumeGroupPatch(destroyed=False),
+            )
+        if res.status_code != 200:
             module.fail_json(
-                msg="Recovery of volume group {0} failed.".format(module.params["name"])
+                msg="Recovery of volume group {0} failed. Error: {1}".format(
+                    module.params["name"], res.errors[0].message
+                )
             )
 
     module.exit_json(changed=changed)
 
 
 def eradicate_vgroup(module, array):
     """Eradicate Volume Group"""
+    api_version = array.get_rest_version()
     changed = True
     if not module.check_mode:
-        try:
-            array.eradicate_vgroup(module.params["name"])
-        except Exception:
+        if LooseVersion(PRIORITY_API_VERSION) <= LooseVersion(api_version):
+            res = array.delete_volume_groups(
+                names=[module.params["name"]], context_names=[module.params["context"]]
+            )
+        else:
+            res = array.delete_volume_groups(names=[module.params["name"]])
+        if res.status_code != 200:
             module.fail_json(
-                msg="Eradicating vgroup {0} failed.".format(module.params["name"])
+                msg="Eradicating vgroup {0} failed. Errors: {1}".format(
+                    module.params["name"], res.errors[0].message
+                )
             )
     module.exit_json(changed=changed)
 
 
 def delete_vgroup(module, array):
     """Delete Volume Group"""
+    api_version = array.get_rest_version()
     changed = True
     if not module.check_mode:
-        try:
-            array.destroy_vgroup(module.params["name"])
-        except Exception:
+        if LooseVersion(PRIORITY_API_VERSION) <= LooseVersion(api_version):
+            res = array.patch_volume_groups(
+                names=[module.params["name"]],
+                context_names=[module.params["context"]],
```

Comment:
its causing array's in lower purity version to fail as it won't have context, please check.